### PR TITLE
mem cache put function fixed

### PR DESCRIPTION
**Pull request by JessonChan**
when a expire duration==0,it means forever
https://github.com/astaxie/beego/issues/1259

### CODE REVIEW

**Comment by JessonChan** [comment]
@fuxiaohei 
plz review this pr.

**Comment by JessonChan** [comment]
yeap,but i think it's ok.
have a look at file-mem
https://github.com/JessonChan/beego/blob/0b390912921919df1b75894a9ff06b6b2665cc75/cache/file.go#L150

**Comment by fuxiaohei** [comment]
No need to follow file cache. There is no gc in file-mem.

And maybe if `item.Expired == 0`, reduce api calls in 

https://github.com/JessonChan/beego/blob/0b390912921919df1b75894a9ff06b6b2665cc75/cache/memory.go#L58

    if itm, ok := bc.items[name]; ok && itm.Expired > 0{

**Comment by JessonChan** [comment]
thanks @fuxiaohei, please review again.

File: cache/memory.go [hunk 1]
```diff
@@ -52,11 +60,10 @@ func NewMemoryCache() Cache {
 // Get cache from memory.
 // if non-existed or expired, return nil.
 func (bc *MemoryCache) Get(name string) interface{} {
-	bc.lock.RLock()
-	defer bc.lock.RUnlock()
+	bc.RLock()
+	defer bc.RUnlock()
 	if itm, ok := bc.items[name]; ok {
-		if (time.Now().Unix() - itm.Lastaccess.Unix()) > itm.expired {
-			go bc.Delete(name)
```
Comment (JessonChan) [inline cache/memory.go]: as commit log says.
the memory gc goroutine will do the delele,no more goroutine here.

**Comment by JessonChan** [comment]
@fuxiaohei typo fixed

File: cache/memory.go (hunk 1 above)
Comment (fuxiaohei) [inline cache/memory.go]: LGTM

**Comment by fuxiaohei** [comment]
https://github.com/JessonChan/beego/blob/develop/cache/memory.go#L65

it can:

```go
if itm, ok := bc.items[name]; ok && !itm.isExpired() {
    return itm.val
```

**Comment by JessonChan** [comment]
@fuxiaohei 
i think the style  is not so readable.
hmmm,i prefer my version .

**Comment by fuxiaohei** [comment]
LGTM

maybe I prefer compact style

**Pull request merged**